core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ZoneService.java

```diff
@@ -90,7 +103,53 @@ public List<Zone> findAll(Predicate<Zone> filter) {
 
   @PreAuthorize("hasPermission(#zone, 'DELETE')")
   public void delete(Zone zone) {
-    throw new UnsupportedOperationException("Zone deletion not currently supported.");
+    handlerService.handleDelete(zone);
+    streamBindingView
+      .findAll(sb -> sb.getKey().getInfrastructureZone().equals(zone.getKey().getName()))
+      .findAny()
+      .ifPresent(sb -> { throw new IllegalStateException("Zone is used in stream: " + sb.getKey().getStreamKey()); });
+
+    consumerBindingView
+      .findAll(cb -> cb.getKey().getInfrastructureZone().equals(zone.getKey().getName()))
+      .findAny()
+      .ifPresent(cb -> { throw new IllegalStateException("Zone is used in consumer binding: " + cb.getKey()); });
+
+    producerBindingView
+      .findAll(pb -> pb.getKey().getInfrastructureZone().equals(zone.getKey().getName()))
+      .findAny()
+      .ifPresent(pb -> { throw new IllegalStateException("Zone is used in producer binding: " + pb.getKey()); });
+
+    processBindingView
+      .findAll(pb -> isZoneUsedInProcessBinding(zone, pb))
+      .findAny()
+      .ifPresent(pb -> { throw new IllegalStateException("Zone is used in process binding: " + pb.getKey()); });
+
+    processView
+      .findAll(p -> p.getZones().stream().toList().contains(zone.getKey()))
+      .findAny()
+      .ifPresent(p -> { throw new IllegalStateException("Zone is used in process: " + p.getKey()); });
+
+    infrastructureView
+      .findAll(infra -> infra.getKey().getZone().equals(zone.getKey().getName()))
+      .findAny()
+      .ifPresent(infra -> { throw new IllegalStateException("Zone is used in infrastructure: " + infra.getKey()); });
+
+    zoneRepository.delete(zone);
   }
 
+  private boolean isZoneUsedInProcessBinding(Zone zone, ProcessBinding processBinding) {
+    return processBinding.getKey().getInfrastructureZone().equals(zone.getKey().getName()) ||
+      isZoneUsedInProcessBindingOutput(zone, processBinding) ||
+      isZoneUsedInProcessBindingInput(zone, processBinding);
+  }
+
+  private boolean isZoneUsedInProcessBindingOutput(Zone zone, ProcessBinding processBinding) {
+    return processBinding.getOutputs().stream().map(o -> o.getStreamBindingKey().getInfrastructureZone()).toList()
+      .contains(zone.getKey().getName());
```

Comment:
Is there any need for the `toList.contains()`? I think we can use `anyMatch()`?

Reply:
Yes, it's not required to convert to list unnecessarily. I'll replace it with `anyMatch()`

Reply:
Done